### PR TITLE
[GCAL] Store conference data

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary
Improve online meetings parsing by using a new field called `Conference`.

It will store data there if the location is an URL and if the event coming from the remote has proper conference data stored in their result. (only did for google calendar)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53868

### CODE REVIEW

File: server/mscalendar/views/calendar.go [hunk 1]
```diff
@@ -157,11 +152,22 @@ func RenderEventAsAttachment(event *remote.Event, timezone string, options ...Op
...
-		if isKnownMeetingURL(event.Location.DisplayName) {
-			titleLink = event.Location.DisplayName
+		title := "Meeting URL"
+		if event.Conference.Application != "" {
+			title = event.Conference.Application
 		}
+
+		fields = append(fields, &model.SlackAttachmentField{
+			Title: title,
+			Value: event.Conference.URL,
```
Comment (mickmister) [inline server/mscalendar/views/calendar.go:168]: 0/5, may be error-prone to do it my suggested way

```suggestion
			Value: titleLink,
```

File: server/remote/gcal/get_default_calendar_view.go [hunk 2]
```diff
@@ -83,8 +84,26 @@ func convertGCalEventToRemoteEvent(event *calendar.Event) *remote.Event {
...
+			conference.Application = event.ConferenceData.ConferenceSolution.Name
+		}
+	} else if utils.IsURL(event.Location) {
+		conference = &remote.Conference{
+			URL: event.Location,
+		}
+	}
+
+	if !utils.IsURL(event.Location) {
+		location = &remote.Location{
```
Comment (mickmister) [inline server/remote/gcal/get_default_calendar_view.go:104]: If I understand correctly, the output event's `event.Conference` and `event.Location` should never be defined at the same time? Just want to understand the implications of this change

File: server/remote/gcal/get_default_calendar_view_test.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+				return evt
+			},
+			Check: func(t *testing.T, event *remote.Event) {
+				require.Equal(t, "https://example.com/meeting", event.Conference.URL)
+				require.Equal(t, "example", event.Conference.Application)
+				require.Empty(t, event.Location)
+			},
+		},
+		{
+			Name: "location url used as conference if not conference data is present",
```
Comment (mickmister) [inline server/remote/gcal/get_default_calendar_view_test.go:119]:
```suggestion
			Name: "location url used as conference if no conference data is present",
```

File: server/remote/gcal/get_default_calendar_view_test.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	return calendar.Event{
+		Summary: "test summary",
+		Start: &calendar.EventDateTime{
+			DateTime: "2023-08-01T00:01:02Z",
+			TimeZone: "UTC",
+		},
+		End: &calendar.EventDateTime{
+			DateTime: "2023-08-01T00:01:02Z",
+			TimeZone: "UTC",
+		},
```
Comment (mickmister) [inline server/remote/gcal/get_default_calendar_view_test.go:22]: Just to be more realistic, can we have the end time be later?

File: server/remote/gcal/get_default_calendar_view_test.go [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+				require.Empty(t, event.Conference.Application)
+				require.Empty(t, event.Location)
+			},
+		},
+	} {
+		t.Run(tc.Name, func(t *testing.T) {
+			event := tc.In()
+			result := convertGCalEventToRemoteEvent(&event)
+			tc.Check(t, result)
+		})
```
Comment (mickmister) [inline server/remote/gcal/get_default_calendar_view_test.go:136]: Nice, simple test :+1:

**Review by mickmister — APPROVED** [review]

File: server/remote/gcal/get_default_calendar_view_test.go (hunk 4 above)
Comment (mickmister) [reply server/remote/gcal/get_default_calendar_view_test.go:22]: Did you want to implement this change?

Comment (fmartingr) [reply server/remote/gcal/get_default_calendar_view_test.go:22]: I recall doing it, that's why I resolved this... 🤔

**Review by mickmister — APPROVED** [review]

**Pull request merged**